### PR TITLE
Speed up QGIS compilation by removing all Qt Module-wide includes

### PR DESCRIPTION
**Pull request by nyalldawson**
Refs https://www.kdab.com/beware-of-qt-module-wide-includes/

### CODE REVIEW

File: src/core/expression/qgsexpressionutils.h [hunk 1]
```diff
@@ -28,6 +28,8 @@
 #include "qgsrelationmanager.h"
 #include "qgsvectorlayer.h"
 
+#include <QThread>
```
Comment (m-kuhn) [inline src/core/expression/qgsexpressionutils.h:31]: That looks like a false positive by case sensitivity?

**Comment by roya0045** [comment]
I've been wondering if there was a way to do dynamic testing and only validate elements that would be affected by a code change. Now I know this could be complex to do if such a thing must be done in qgis, but I feel like this could save a good amount of CI time. 

Something as basic as testing all when core is change, testing app and gui when app is change, and testing gui when it is changed. Or anything along those line for a rough start. 

Unless there are solutions that can link files and their dependency and filter out the irrelevant tests.

**Comment by nyalldawson** [comment]
(Just to confirm... this does make a HUGE difference to the QGIS compilation speed :partying_face: )

**Comment by nirvn** [comment]
:man: :boom:  :robot:

**Comment by 3nids** [comment]
nice catch!

**Pull request merged**